app/components/trainees/record_actions/view.rb

```diff
@@ -10,6 +10,18 @@ class View < GovukComponent::Base
       def initialize(trainee)
         @trainee = trainee
       end
+
+      def display_actions?
+        allow_defer? || allow_withdraw?
+      end
+
+      def allow_defer?
+        trainee.submitted_for_trn? || trainee.trn_received?
+      end
+
+      def allow_withdraw?
+        allow_defer? || trainee.deferred?
+      end
```

Comment:
Feels like these can be in the model

Reply:
I had considered it, but then I thought, if these methods are not going to be used elsewhere, it's best to keep them out of the model for now.